internal/api/utils.go

```diff
@@ -232,52 +232,89 @@ func UpdateMemoryFromBytesToMiB(jsonData datatypes.JSON) map[string]interface{}
 		amount, ok := memory["amount"].(float64)
 		if ok {
 			memoryInMiB := amount / 1024 / 1024
-			memory["amount"] = math.Trunc(memoryInMiB*100) / 100
-			memory["format"] = "MiB"
-			if memoryInMiB >= 1024 {
+			if math.Abs(memoryInMiB) >= 1024 {
 				memoryInGiB := memoryInMiB / 1024
 				memory["amount"] = math.Trunc(memoryInGiB*100) / 100
 				memory["format"] = "GiB"
+			} else {
+				memory["amount"] = math.Trunc(memoryInMiB*100) / 100
+				memory["format"] = "MiB"
+			}
+		}
+		return nil
+	}
+
+	hasMoreThanThreeDecimals := func(value float64) bool {
+		const decimalPrecision int = 3
+		str := strconv.FormatFloat(value, 'f', -1, 64)
+		decimalPart := strings.Split(str, ".")
+		return (len(decimalPart) > 1) && (len(decimalPart[1]) > decimalPrecision)
+	}
+
+	truncateToThreeDecimalPlaces := func(value float64) float64 {
+		if hasMoreThanThreeDecimals(value) {
+			truncated := math.Trunc(value * 1000) // Pushes decimal by 3 places and then truncates
+			return truncated / 1000
+		}
+		return value
+	}
+
+	convertCPU := func(cpu map[string]interface{}) error {
+		cpuInCores, ok := cpu["amount"].(float64)
+		if ok {
+			if math.Abs(cpuInCores) < 1 {
+				cpuInMillicores := cpuInCores * 1000
+				cpu["amount"] = math.Round(cpuInMillicores) // millicore values are rounded & don't require decimal precision
+				cpu["format"] = "millicores"
+			} else {
+				cpu["amount"] = truncateToThreeDecimalPlaces(cpuInCores)
+				cpu["format"] = "cores"
 			}
 		}
 		return nil
 	}
 
+	/*
+		Recommendation data is available for three periods
+		For each of these actual values will be present in
+		below mentioned dataBlocks > request and limits
+	*/
+
 	for _, period := range []string{"long_term", "medium_term", "short_term"} {
 		intervalData, ok := durationBased[period].(map[string]interface{})
 		if !ok {
 			continue
 		}
 
-		for _, dataBlock := range []string{"current", "config", "variation"} {
+		for _, dataBlock := range []string{"config", "variation"} {
```

Comment:
I think we should not remove `current` from the slice. It was recently added. Thoughts?

Reply:
I believe git/GitHub will auto-resolve this, based on the PR/branch creation times.

Reply:
Ok, we can merge but I strongly feel it won't be auto-resolved.

Reply:
My bad, my rebase would override history, thanks! Change restored